catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -203,27 +217,25 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             if (deleteResponse != null) {
                 List<Metacard> deletedMetacards = deleteResponse.getDeletedMetacards();
 
-                for (Metacard delete : deletedMetacards) {
-                    fileIdMap.delete(ingestedFile.getAbsolutePath());
-                    LOGGER.debug("content item deleted with id = {}", delete.getId());
+                if (safeKey != null) {
+                    fileIdMap.delete(safeKey);
                 }
+                logIds(deletedMetacards, "deleted");
             }
         }
     }
 
-    public void processHeaders(Map<String, Object> headers, StorageRequest storageRequest,
-            File ingestedFile) {
-        Map<String, Serializable> attributeOverrideHeaders = new HashMap<>((Map) headers.get(
-                Constants.ATTRIBUTE_OVERRIDES_KEY));
-        if (!attributeOverrideHeaders.isEmpty()) {
-            storageRequest.getProperties()
-                    .put(Constants.ATTRIBUTE_OVERRIDES_KEY,
-                            (Serializable) attributeOverrideHeaders);
-        }
-        if (headers.containsKey(Constants.STORE_REFERENCE_KEY)) {
-            storageRequest.getProperties()
-                    .put(Constants.STORE_REFERENCE_KEY, ingestedFile.getAbsolutePath());
-        }
+    protected void logIds(List<Metacard> metacards, String action) {
+        LOGGER.debug("content item(s) {} with id = {}",
+                action,
+                metacards.stream()
+                        .map(Metacard::getId)
+                        .filter(Objects::nonNull)
+                        .reduce((s, s2) -> String.format("%s, %s", s, s2)));
```

Comment:
❗️ This will log the `toString()` of the optional returned from `reduce`, giving you outputs like
`content item(s) updated with id = Optional[x, y, z]`
and potentially
`content item(s) updated with id = Optional.empty`.

There are a couple of ways around that, but the easiest is to swap out the reducer with a call to 
`.collect(Collectors.joining(", "));`

That will get you the output you want and handle the empty collection of ids.